File: Python/Daily007_easy/Morse.py
## Morse
##
## challenge #7 (easy)
## http://www.reddit.com/r/dailyprogrammer/comments/pr2xr/2152012_challenge_7_easy/
## 
## 
## sarthak7u@gmail.com
import string
import sys

morse = [".-", "-...", "-.-.","-..",".","..-.", "--.", "....","..",
			".---","-.-",".-..","--","-.", "---", ".--.", "--.-", ".-.",
			"...","-","..-","...-",".--", "-..-","-.--","--.."]
alpha = list(string.ascii_lowercase)

def morse_to_alpha(alpha,morse):
	letters_in_morse = [""]
	user_input = raw_input("Enter a string in morse: ")

	for i in user_input:
		if i == " ":
			letters_in_morse.append("")
			continue
		letters_in_morse[-1] += i 

	# makes a dictionary for morse to alpha
	morse_to_alpha = dict(zip(morse,alpha))
	morse_to_alpha["/"] = " "

	output =[morse_to_alpha[j] for j in letters_in_morse]
	print "".join(output)

def alpha_to_morse(alpha,morse):
	user_input = raw_input("Enter a string in alpha : ")

	# makes a dictionary for alpha to morse
	alpha_to_morse = dict(zip(alpha,morse))
	alpha_to_morse["/"] = " "

	output =[alpha_to_morse[j] for j in user_input]
	print " ".join(output)

# Ask user which type of conversion
choice = int(raw_input("""
Enter 0 for morse to alpha or 
Enter 1 for alpha to morse:\n"""))

if choice == 0:
	morse_to_alpha(alpha,morse)
elif choice == 1:
	alpha_to_morse(alpha,morse)

File: OCaml/Daily007_easy/morse.ml
(*
  Morse Code Decrypt
  #7 (easy)
  https://www.reddit.com/r/dailyprogrammer/comments/pr2xr/2152012_challenge_7_easy/

  sarthak7u@gmail.com
*)

let get_lookup () = 
  let morse_hash = Hashtbl.create 26 in
  let morse_list = [|"a";"b";"c";"d";"e";"f";"g";"h";"i";"j";"k";"l";"m";"n";"o";"p";"q";"r";"s";"t";"u";"v";"w";"x";"y";"z"|] in
  let morse_code = [|"._";"_...";"_._."; "_.."; "."; ".._.";
  "__."; "...."; ".."; ".___"; "_._"; "._.."; "__"; "_.";
  "___"; ".__."; "__._"; "._."; "..."; "_"; ".._"; "..._";
  ".__"; "_.._"; "_.__"; "__.."|] in

  let i = ref 0 in
  while !i < 26 do
    Hashtbl.add morse_hash morse_code.(!i) morse_list.(!i);
  i := !i + 1
  done;
  morse_hash

let rec decrypt_morse query =
  let l = String.split_on_char ' ' query in
  let hsh = get_lookup() in

  match l with
    | [] -> query
    | [h] -> Hashtbl.find hsh h
    | h::t -> Hashtbl.find hsh h ^ decrypt_morse (String.concat "" t )


let () =
  let decrypted = decrypt_morse(".... .") in
  print_string decrypted;
  print_newline ();

